src/api/uploads/FolderUploadNode.js

```diff
@@ -96,34 +106,58 @@ class FolderUploadNode {
     createAndUploadFolder = async (errorCallback: Function, isRoot: boolean) => {
         await this.buildCurrentFolderFromEntry();
 
+        let errorEncountered = false;
+        let errorCode = '';
         try {
             const data = await this.createFolder();
             this.folderId = data.id;
         } catch (error) {
+            errorEncountered = true;
+            errorCode = error.code;
             // @TODO: Handle 429
             if (error.code !== ERROR_CODE_ITEM_NAME_IN_USE) {
-                errorCallback(error);
-                return;
+                if (!isRoot) {
+                    // If this is a child folder of the folder being uploaded, this errorCallback will set
+                    // an error message on the root folder being uploaded. Set a generic messages saying that a
+                    // child has caused the error. The child folder will be tagged with the error message in
+                    // the call to this.addFolderToUploadQueue below
+                    errorCallback({ code: ERROR_CODE_CHILD_FOLDER_FAILED_UPLOAD });
+                } else {
+                    errorCallback(error);
+                }
+            } else {
+                this.folderId = error.context_info.conflicts[0].id;
             }
-
-            this.folderId = error.context_info.conflicts[0].id;
         }
 
         // The root folder has already been added to the upload queue in ContentUploader
         if (isRoot) {
             return;
         }
 
-        this.addFolderToUploadQueue([
-            {
-                extension: '',
-                name: this.name,
-                status: STATUS_COMPLETE,
-                isFolder: true,
-                size: 1,
-                progress: 100,
-            },
-        ]);
+        const folderObject: {
```

Comment:
const folderObject: UploadItem = {...}

Reply:
This isn't yet an `UploadItem`. This object is passed up and created as an `UploadItem` in a higher level (we don't have access to the required API property at this lower level).